### PR TITLE
[UXP-3353] Handle slice selection in the NGS view

### PR DESCRIPTION
**Pull request by andrejak**
This PR sets up the NGS view for dashboard (or other integration). We now handle the slice selection within the component because that makes more sense in the context of the full offer flow. 
(This looks like a lot of code changes but mostly the code was moved from DuffelNGSView into NGSTable and CSS split up and renamed)

A couple other small changes:
- add a header and breadcrumbs to the ngs view so that the user knows what slice they're dealing with and can go back in their selection (slightly different from design since we're not in Links)
- removed conditions from the offerslice modal as they don't necessarily apply to all the offers in the ngs row
- adjusted some css so it looks better in the dashboard

Screenshot 2024-02-01 at 15 47 47

### CODE REVIEW

File: src/components/DuffelNGSView/NGSTable.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    setIsOfferSliceModalOpen(false);
+    setRows(groupOffersForNGSView(offers, sliceIndex, previousSliceKeys));
+  }, [previousSliceKeys]);
+
+  React.useEffect(() => {
+    if (sortShelf) {
+      const sortedRows = sortNGSRows(rows, sortShelf, sortDirection);
+      setRows(sortedRows);
+    }
+  }, [sortShelf, sortDirection]);
```
Comment (igorp1) [inline src/components/DuffelNGSView/NGSTable.tsx:106]: I don't think we need this? Sorted rows is the output of some state, but itself doesn't need to be persisted, right? 
Something like this I guess?
```suggestion
      const rows = groupOffersForNGSView(offers, sliceIndex, previousSliceKeys)
      const sortedRows = sortShelf ? rows : sortNGSRows(rows, sortShelf, sortDirection); 
```
And then use sortedRows below?

File: src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts [hunk 2]
```diff
@@ -13,17 +13,35 @@ const getSliceKey = (slice: OfferSliceWithNGS): string => {
...
+        if (sliceKey !== getNGSSliceKey(offer.slices[index])) {
+          match = false;
+          break;
+        }
+      }
+      if (match) {
+        filteredOffers.push(offer);
+      }
+    }
+  }
```
Comment (igorp1) [inline src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts:33]: Clear function! But does the name represent it accurately? Maybe should it be `filterOffersThatMatchCurrentSlice` or something like that?

Comment (andrejak) [reply src/components/DuffelNGSView/lib/group-offers-for-ngs-view.ts:33]: Well, this function got quite long, and the second part does the grouping... I will pull out the part you highlighted into its own function called `filterOffersThatMatchCurrentSlice`.

File: src/components/DuffelNGSView/NGSTable.tsx (hunk 1 above)
Comment (andrejak) [reply src/components/DuffelNGSView/NGSTable.tsx:106]: Yeah you're right, not sure why I added that in the first place :/

**Review by igorp1 — APPROVED** [review]
Nice one! Thank you for the updates :)

**Pull request merged**